### ISSUE TITLE
Pull query AST into separate library

### ISSUE THREAD

**Issue by 3noch** [agentm/project-m36#141]
I just realized that with very little work I could have my GHCJS front-end build up Project:M36 queries directly. It could then send them to a websocket on my server and have get the results directly. All the serialization/deserialization would be done automatically. I know Project:M36 already has a websocket server, but this would allow me to do 2 key things:

  1. Generate the AST *on the client directly*
  2. Rely on my applications authentication (which happens via JWT in a header)

**Comment by agentm** [comment]
Indeed, some sort of GHCJS integration would likely be beneficial. It could even be possible to build the entire relational algebra engine in GHCJS (without persistence, of course). 

However, for broad appeal, I chose to make a simple JS library. I was thinking that the JS library could be improved to allow the user to build up a JS-style ADT, in the short-term at least.

**Comment by 3noch** [comment]
Having the JS library is certainly valuable. I'm only advocating for yet another option, GHCJS.
